lib/net.c

```diff
@@ -1391,13 +1448,42 @@ start_iface(struct gatekeeper_if *iface, unsigned int num_attempts_link_get)
 			&num_succ_ports, num_attempts_link_get);
 		if (ret < 0)
 			goto stop_partial;
+
+		/*
+		 * If we try to update/get the RSS hash configuration before
+		 * the start of the NICs, no meaningful operations will be
+		 * done even the return values indicate no errors.
+		 *
+		 * After checking the source code of DPDK library,
+		 * it turns out that RSS is disabled in the MRQC register
+		 * before we start the NICs.
+		 *
+		 * Only after the NICs start, we can check whether the RSS hash
+		 * is configured correctly or not.
+		 */
+		if (check_port_rss_key_update(iface, iface->ports[i])) {
+			G_LOG(ERR,
+				"net: port %hu (%s) on the %s interface does get the RSS hash key updated correctly\n",
+				iface->ports[i], iface->pci_addrs[i],
+				iface->name);
+			ret = -1;
+			goto stop_partial;
+		}
 	}
 
 	/* Bonding port(s). */
 	if (iface_bonded(iface)) {
 		ret = start_port(iface->id, NULL, num_attempts_link_get);
 		if (ret < 0)
 			goto stop_partial;
+
+		if (check_port_rss_key_update(iface, iface->id)) {
+			G_LOG(ERR,
```

Comment:
Since we already check per port above, do we really need to check for the bonded interface?

Reply:
I found there is a comment ``Bonded port inherits RSS and offload settings from the slave ports added to it`` in ``lib/net.c``, so I think we can drop this check.